### ISSUE TITLE
GPU runtime ignores Graph execution error with `sparse_categorical_crossentropy`

### ISSUE THREAD

**Issue by rajdeep-biswas** [googlecolab/colabtools#2967]
**Describe the current behavior**
Setting output layer units to be a number that is lower than possible class labels, should make `sparse_categorical_crossentropy` throw a `Graph execution error` at the fit() step - which works as expected on normal runtime, however, switching the runtime to GPU, the error is ignored and training proceeds as normal (with nan loss and 0 accuracy).

**Describe the expected behavior**
GPU runtime should throw the same error as well instead of letting training proceed. I have created a well-documented notebook that can demonstrate / reproduce this issue and have linked it below.

**What web browser you are using**
Chrome Version 103.0.5060.134 (Official Build) (x86_64).

**Additional context**
[Link to a minimal, public, self-contained notebook that reproduces this issue](https://colab.research.google.com/github/rajdeep-biswas/sparse-categorical-crossentropy/blob/master/sparcat.ipynb).

**Comment by craigcitro** [comment]
This looks like a TensorFlow (or maybe Keras?) issue -- could you try filing a bug upstream?
